--- qemu-2.5.1/cpu-exec.c.orig  2016-04-27 03:44:02.035401672 +0900
+++ qemu-2.5.1/cpu-exec.c       2016-04-27 03:44:18.202976902 +0900
@@ -31,6 +31,7 @@
 #include "hw/i386/apic.h"
 #endif
 #include "sysemu/replay.h"
+#include "../patches/afl-qemu-cpu-inl.h"

 /* -icount align implementation. */

@@ -289,6 +290,8 @@
     /* if no translated code available, then translate it now */
     tb = tb_gen_code(cpu, pc, cs_base, flags, 0);

+    AFL_QEMU_CPU_SNIPPET1;
+
 #ifdef CONFIG_USER_ONLY
     mmap_unlock();
 #endif
@@ -499,6 +502,9 @@
                     next_tb = 0;
                     tcg_ctx.tb_ctx.tb_invalidated_flag = 0;
                 }
+
+                AFL_QEMU_CPU_SNIPPET2;
+
                 if (qemu_loglevel_mask(CPU_LOG_EXEC)) {
                     qemu_log("Trace %p [" TARGET_FMT_lx "] %s\n",
                              tb->tc_ptr, tb->pc, lookup_symbol(tb->pc));
