--- qemu-2.5.1/linux-user/elfload.c.orig        2016-04-27 03:48:33.832291993 +0900
+++ qemu-2.5.1/linux-user/elfload.c     2016-04-27 03:48:39.484144734 +0900
@@ -28,6 +28,8 @@

 #define ELF_OSABI   ELFOSABI_SYSV

+extern abi_ulong afl_entry_point, afl_start_code, afl_end_code;
+
 /* from personality.h */

 /*
@@ -1916,6 +1918,8 @@
     info->brk = 0;
     info->elf_flags = ehdr->e_flags;

+    if (!afl_entry_point) afl_entry_point = info->entry;
+
     for (i = 0; i < ehdr->e_phnum; i++) {
         struct elf_phdr *eppnt = phdr + i;
         if (eppnt->p_type == PT_LOAD) {
@@ -1949,9 +1953,11 @@
             if (elf_prot & PROT_EXEC) {
                 if (vaddr < info->start_code) {
                     info->start_code = vaddr;
+                    if (!afl_start_code) afl_start_code = vaddr;
                 }
                 if (vaddr_ef > info->end_code) {
                     info->end_code = vaddr_ef;
+                    if (!afl_end_code) afl_end_code = vaddr_ef;
                 }
             }
             if (elf_prot & PROT_WRITE) {
